.zuul.yaml

```diff
@@ -10,6 +10,8 @@
       golangci_lint_version: 1.59.1
       go_version: 1.22.4
       golangci_lint_options: --timeout 5m
+    provides:
```

Comment:
you do not really need provides/requires in jobs here. In Zuul it means something different and used for requirements between different projects and not between jobs in the same project (https://zuul-ci.org/docs/zuul/latest/config/job.html#attr-job.requires)

Reply:
Misunderstood the usage of it, we were thinking of setting an order, in witch the jobs run, to save on nodes. 

We don't care if tests succeed, if the linting fails. We don't care for building, if testing fails, etc.

Is there a way of doing such a workflow?

Reply:
yes, there is: https://github.com/opentelekomcloud-infra/system-config/blob/main/zuul.d/project.yaml#L68 or https://opendev.org/openstack/codegenerator/src/branch/master/zuul.d/openapi.yaml#L282 as example (https://zuul-ci.org/docs/zuul/latest/config/job.html#attr-job.dependencies for reference). Building it this way, though, you make your CI longer. You generally should not do this unless jobs are very long running.

It makes much more sense to use pods for running lightweight jobs (https://github.com/SovereignCloudStack/zuul-scs-jobs/blob/main/zuul.d/jobs.yaml#L35) with `nodeset: pod-fedora-40`. Such jobs are being executed in K8 and do not require nodes to be provisioned. There are few cases where this should not be used:
- image building (docker in docker never worked good)
- very heavy jobs (requiring more cpu/mem).
- very long running jobs (networking bugs on k8 cause Zuul loose control over those)

Reply:
That seems nice, instead of waiting for the nodes.

Is there a list, of node sets and labels, that are currently usable, documented somewhere?

Reply:
https://github.com/SovereignCloudStack/zuul-scs-jobs/blob/main/zuul.d/nodesets.yaml

Reply:
so do you plan to drop provides/requires? The only sense to use those would be in combination between status-page-api and status-page-web to build dependencies between projects

Reply:
dropped the provides and requires in jobs